File: golang-leetcode/70.go
package main

import "fmt"

func main() {
	fmt.Println("climbStairs(3) = ", climbStairs(3))
}

func climbStairs(n int) int {
	if n < 3 {
		return n
	}
	f1, f2 := 1, 2
	for i := 3; i <= n; i++ {
		f1, f2 = f2, f1+f2
	}
	return f2
}

File: cpp-leetcode/70.cpp
#include <iostream>
#include <vector>
#include <utility>
using namespace std;

class Solution {
public:
  int climbStairs(int n) {
    if (n < 3) {
      return n;
    }
    int f1 = 1;
    int f2 = 2;
    for (int i = 2; i < n; i++) {
      int next = f1 + f2;
      f1 = f2;
      f2 = next;
    }
    return f2;
  }
};

int main() {
  Solution o;
  int result = o.climbStairs(3);
  printf("result = %d\n", result);
  return 0;
}
